src/main/java/io/syndesis/qe/marketplace/openshift/OpenShiftService.java

```diff
@@ -203,4 +212,75 @@ public void patchGlobalSecrets(String pullSecretContent) {
+        CustomResourceDefinitionContext crds = new CustomResourceDefinitionContext.Builder()
+                .withVersion("v1")
+                .withKind("PackageManifest")
+                .withGroup("packages.operators.coreos.com")
+                .withScope("Namespaced")
+                .withPlural("packagemanifests")
+                .build();
+
+        final ObjectMapper mapper = new ObjectMapper();
+
+        openShiftClient.customResource(crds).list("openshift-marketplace").entrySet()
+                .stream().filter(entry -> "items".equals(entry.getKey())).findFirst().ifPresent( entry -> {
+                        found.set(((List<Map>) entry.getValue()).stream().collect(
+                                Collectors.groupingBy(pack -> (String) ((Map) pack.get("status")).get("catalogSource"),
+                                        HashMap::new,
+                                        Collectors.mapping(item -> mapper.convertValue(item.get("status"), PackageManifest.class), Collectors.toList())
+                                )));
+
+                    }
+                );
+
+        return found.get();
+    }
+
+    /**
+     * Retrieve the default channel of the operator with given name in given catalog source.
+     * @param source String, the catalog source
+     * @param operatorName String, the operator name as found in the catalog
+     * @param refreshCatalog boolean, if the catalog should be refreshed
+     * @return the {@link PackageChannel} found or {@link IllegalArgumentException} if not found
+     */
+    public PackageChannel getDefaultChannel(final String source, final String operatorName, boolean refreshCatalog) {
+        return getChannel(source, operatorName, null, refreshCatalog);
+    }
+
+    /**
+     * Retrieve the channel by name, of the operator with given name in given catalog source.
+     * @param source String, the catalog source
+     * @param operatorName String, the operator name as found in the catalog
+     * @param channelName String, the channel name as found in the catalog
+     * @param refreshCatalog boolean, if the catalog should be refreshed
+     * @return the {@link PackageChannel} found or {@link IllegalArgumentException} if not found
+     */
+    public PackageChannel getChannel(final String source, final String operatorName, final String channelName, boolean refreshCatalog) {
+        getCatalog(refreshCatalog);
+        PackageManifest found = catalog.get(source).stream().filter(pack -> operatorName.equals(pack.getPackageName()))
+                .findFirst().orElseThrow(() -> new IllegalArgumentException("Operator " + operatorName + " not found"));
+        String channelToUse = channelName != null ? channelName : found.getDefaultChannel();
+        return found.getChannels().stream().filter(ch -> channelToUse.equals(ch.getName())).findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Channel " + channelToUse + " not found"));
```

Comment:
Same thing here: JSONObjects might end up more readable

Reply:
no json are managed here, the method getCatalg() returns an object (Map), here the stream APIs has been used just to search/filter the result

Reply:
You can create a JSONObject as an "abstraction" over the map so you can easily traverse the structure http://javadox.com/org.json/json/20151123/org/json/JSONObject.html#JSONObject-java.util.Map-

Reply:
no json are managed here, the method getCatalg() returns an object (Map), here the stream APIs has been used just to search/filter the result